File: learn/src/main/java/com/example/jojo/learn/MainActivity.java
package com.example.jojo.learn;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.jojo.learn.customview.MyTextView;
import com.example.jojo.learn.customview.SportStepView;

public class MainActivity extends AppCompatActivity {

    private MyTextView mytextview;
    private float mCurrentStep;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        mytextview = findViewById(R.id.mytextview);
        final SportStepView sportStepView = findViewById(R.id.sportstepview);
        //当前实际的步数
        mCurrentStep = 9709;
        sportStepView.startCountStep(mCurrentStep);

    }

    //    public void textLayoutLeft(View v) {
//        mytextview.reLayoutText(0);
//    }
//    public void textLayoutCenter(View v) {
//        mytextview.reLayoutText(1);
//    }
}

File: app/src/main/java/com/example/jojo/mvp_kotlin/MainActivity.kt
package com.example.jojo.mvp_kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.TextView
import com.example.jojo.mvp_kotlin.net.retrofit.RxSchedulers
import com.zongxueguan.naochanle_android.retrofitrx.RetrofitRxManager
import io.reactivex.observers.DisposableObserver

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val tv = findViewById<TextView>(R.id.tv);
        RetrofitRxManager.getRequestService().getWeather("北京")
                .compose(RxSchedulers.io_main())
                .subscribeWith(object : DisposableObserver<Any>() {
                    override fun onNext(t: Any) {
                        tv.text = t.toString()
                        Log.e("TAG", "onNext" + t)
                    }

                    override fun onError(e: Throwable) {
//                        onErrorExpected a string but was BEGIN_OBJECT at line 1 column 2 path $
                        Log.e("TAG", "onError" + e.message)
                    }

                    override fun onComplete() {
                        Log.e("TAG", "onComplete")
                    }
                })
    }
}
